### PR TITLE
Addressing PodSecurity violation warnings:

### LINKED ISSUE #2683: Restricted PodSecurity violation from deployments/statefulsets
Running with a restricted `knative-eventing` Namespace:

```yaml
apiVersion: v1
kind: Namespace
metadata:
  name: knative-eventing
  labels:
    eventing.knative.dev/release: devel
    app.kubernetes.io/version: devel
    app.kubernetes.io/name: knative-eventing
    pod-security.kubernetes.io/enforce: restricted
    pod-security.kubernetes.io/warn: restricted
    pod-security.kubernetes.io/audit: restricted 
```

Results in warnings, like (here for `Deployment`s of the `kafka-broker-dispatcher` and `kafka-broker-receiver`):
```
Warning: would violate PodSecurity "restricted:latest": unrestricted capabilities (container "kafka-broker-dispatcher" must set securityContext.capabilities.drop=["ALL"]), seccompProfile (pod or container "kafka-broker-dispatcher" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
deployment.apps/kafka-broker-dispatcher created
Warning: would violate PodSecurity "restricted:latest": unrestricted capabilities (container "kafka-broker-receiver" must set securityContext.capabilities.drop=["ALL"]), seccompProfile (pod or container "kafka-broker-receiver" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```

Similar for our `StatefulSet`s:
```
Warning: would violate PodSecurity "restricted:latest": unrestricted capabilities (container "kafka-source-dispatcher" must set securityContext.capabilities.drop=["ALL"]), seccompProfile (pod or container "kafka-source-dispatcher" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```

### PR DESCRIPTION
**Pull request by matzew**
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

Fixes #2683 

## Proposed Changes

- adding/updating securityContext, as needed, to allow running as 'restricted' standard.
- adding seccompProfile

PR references from knative/eventing repo:
* https://github.com/knative/eventing/pull/5863
* https://github.com/knative/eventing/pull/6533

**Release Note**

```release-note
All core Knative Kafka Pods should now be able to run in the restricted pod security standard profile.
```

**Docs**

### CODE REVIEW

**Comment by matzew** [comment]
```
  containerStatuses:
  - image: ghcr.io/openzipkin/zipkin:2
    imageID: ""
    lastState: {}
    name: zipkin
    ready: false
    restartCount: 0
    started: false
    state:
      waiting:
        message: 'container has runAsNonRoot and image has non-numeric user (zipkin),
          cannot verify user is non-root (pod: "zipkin-67f7b9d877-4gc9m_knative-eventing(795c20e4-482e-45fa-9cd6-86815f58a6b3)",
          container: zipkin)'
        reason: CreateContainerConfigError
```

**Comment by matzew** [comment]
/assign @pierDipi I think this is ready now.
I leave the ZIPKIN as is for now - will check on that later ...

File: control-plane/config/post-install/500-storage-version-migrator.yaml [hunk 1]
```diff
@@ -43,3 +43,12 @@ spec:
...
             - "kafkasinks.eventing.knative.dev"
+          securityContext:
+            allowPrivilegeEscalation: false
+            readOnlyRootFilesystem: true
+            runAsNonRoot: true
+            capabilities:
+              drop:
+              - ALL
+            seccompProfile:
+              type: RuntimeDefault
```
Comment (pierDipi) [inline control-plane/config/post-install/500-storage-version-migrator.yaml:54]:
```suggestion
              type: RuntimeDefault

```

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**